tycho-extras/tycho-pomless/src/main/java/org/eclipse/tycho/pomless/TychoFeatureMapping.java

```diff
@@ -33,6 +34,18 @@ public class TychoFeatureMapping extends AbstractXMLTychoMapping {
     private static final String FEATURE_XML = "feature.xml";
     public static final String PACKAGING = "eclipse-feature";
 
+    @Override
+    protected String getPackaging() {
+        return PACKAGING;
+    }
+
+    @Override
+    public float getPriority() {
+        // It is important that this mapping has a higher priority than the 'XmlMapping' provided by polyglot. 
+        // Otherwise it can happen that the the pom.xml reader is used to read the feature.xml which will break the build!
+        return +1;
```

Comment:
Why should this ever happen?

Reply:
It happened in during my testing of this change, when I made the `org.sonatype.maven.polyglot.mapping.XmlMapping` the highest priority mapping, to reflect that pom.xml should always have precedence. I wanted to avoid the corresponding check in `AbstractTychoMapping.locatePom()`. But that attempt failed for the described reason.

Alternatively the XMLMapping could be adjusted to not accept `feature.xml` files. I.e. by replacing ".xml" by "pom.xml" in its `acceptLocationExtensions`.
You can test that already. Just assign the `TychoFeatureMapping` a priority below zero and run the pomless-model test build and you see that incomplete polyglot-poms are generated for features.

Reply:
> Alternatively the XMLMapping could be adjusted to not accept `feature.xml` files. I.e. by replacing ".xml" by "pom.xml" in its `acceptLocationExtensions`.

This must be fixed then, we can't know what things has what priority and a mapping should only accept files that it really can handle.

Reply:
```
 public XmlMapping() {
    super(null);
    setPomNames("pom.xml");
    setAcceptLocationExtensions(".xml", ".pom");
    setAcceptOptionKeys("xml:4.0.0");
  }
```

I'm not sure if maybe something else is broken here, according to the code, it should only ever accept `pom.xml` and return for all other cases a null (meaning : there is no file I can handle)

Reply:
The problem is the ".xml" extension.
When `PolyglotModelManager.getReaderFor()` is called it asks each mapping (in descending priority order) if it accepts the given options. And `MappingSupport.accept()` falsely returns true for a `feature.xml`:
```
  public boolean accept(final Map<String, ?> options) {
    if (options != null) {
      for (String key : getAcceptOptionKeys()) {
        if (options.containsKey(key)) {
          return true;
        }
      }
      String location = getLocation(options);
      if (location != null) {
        for (String ext : getAcceptLocationExtensions()) {
          if (location.endsWith(ext)) {
            return true;
          }
        }
      }
    }
    return false;
  }
```
The XML-model in the feature.xml file (usually) does not have the "xml:4.0.0" options-key set, but its location ends with `feature.xml`. In fact that Mapping accepts itself to provide a reader for any XML-file. That's the issue.
Until now we are (probably) lucky that the `XMLMapping` is always sorted behind the `TychoFeatureMapping` in `PolyglotModelManager.getSortedMappings()`, but I assume that this is an implementation detail of plexus.

Reply:
I think I fixes a while back some thing regarding the handling already, this defiantly does not looks correct and the xml mapping might need to override this.

This seems to be a very simplifed case because orignianlly polyglot was written to provide a json or yam and then the filtering by extension might work but in general this seems not sufficient, at laest the XML mapping then might better have a very low priority instead so it is used as last resort.